lib/mongoose_push/service/apns/supervisor.ex

```diff
@@ -0,0 +1,70 @@
+defmodule MongoosePush.Service.APNS.Supervisor do
+  @moduledoc """
+  APNS module supervising Sparrow's PoolSupervisor and APNS State
+  """
+  use Supervisor, id: :apns_supervisor
+  require Logger
+
+  @spec start_link([Keyword.t()]) :: Supervisor.on_start()
+  def start_link(arg) do
+    Supervisor.start_link(__MODULE__, arg)
+  end
+
+  @impl true
+  def init(pool_configs) do
+    {dev_list, prod_list} = List.foldl(pool_configs, {[], []}, &convert_pool_to_sparrow/2)
+    sparrow_config = [{:dev, dev_list}, {:prod, prod_list}]
```

Comment:
I would suggest moving this whole translation to external function, to keep the init clean and simple.